recipes/config.rb

```diff
@@ -1,21 +1,60 @@
-template '/etc/default/twemproxy.override' do
-  source 'default.erb'
-  mode '0644'
-  owner 'root'
-  group 'root'
-  variables :args => "--conf-file=#{node['twemproxy']['config_file']}"
-  notifies :reload, 'service[twemproxy]'
-end
+#
+# Cookbook Name:: twemproxy
+# Recipe:: config
+#
+# Copyright 2014, Virender Khatri
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-directory ::File.dirname(node['twemproxy']['config_file']) do
-  recursive true
+template node['twemproxy']['service_config_file'] do
```

Comment:
@guilhem i would agree generally, but in this case different attributes makes sense. like, i would have to add `--conf-file=#{node['twemproxy']['config_file']` and it defeat the purpose if using chef role (json) because reference to conf file would not be simple. also keeping single attribute `:args` does not allow me to use chef roles to differ the service attributes across environments. could you please elaborate why it is a bad idea, may be i am missing the obvious.